From cd6d5cb1cfe03d4fcef8fd38decd04765c19890a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=B4me=20Martinez?= <jerome@mediaarea.net>
Date: Tue, 28 Sep 2021 08:48:35 +0200
Subject: [PATCH] x Fix compilation issue with e.g. musl

---
 Source/MediaInfo/MediaInfo_Internal.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Source/MediaInfo/MediaInfo_Internal.cpp b/Source/MediaInfo/MediaInfo_Internal.cpp
index 03358c7c9..3bb3c24ff 100644
--- a/Source/MediaInfo/MediaInfo_Internal.cpp
+++ b/Source/MediaInfo/MediaInfo_Internal.cpp
@@ -72,11 +72,13 @@
         #undef Yield
         #undef max
     }
-    #elif defined(_POSIX_PRIORITY_SCHEDULING)
-        #include <sched.h>
+    #else
         #include <unistd.h>
         #include <signal.h>
-    #endif //_POSIX_PRIORITY_SCHEDULING
+        #if defined(_POSIX_PRIORITY_SCHEDULING) // Note: unistd.h must be included first
+            #include <sched.h>
+        #endif //_POSIX_PRIORITY_SCHEDULING
+    #endif
     #include <ctime>
 #endif
 using namespace ZenLib;
