diff --git a/Makefile b/Makefile
index 2b2c3de..ce73e1e 100644
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,7 @@ DBM_INCLUDE:=$(DESTDIR)/usr/include
 
 SHARED_LIBS = y
 
-DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
+DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\" -DDATADIR=\"$(DATADIR)\"
 ifneq ($(SHARED_LIBS),y)
 DEFINES+= -DNO_SHARED_LIBS
 endif
diff --git a/netem/Makefile b/netem/Makefile
index ba4c5a7..cb197af 100644
--- a/netem/Makefile
+++ b/netem/Makefile
@@ -23,9 +23,9 @@ stats: stats.c
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)$(LIBDIR)/tc
+	mkdir -p $(DESTDIR)$(DATADIR)/tc
 	for i in $(DISTDATA); \
-	do install -m 644 $$i $(DESTDIR)$(LIBDIR)/tc; \
+	do install -m 644 $$i $(DESTDIR)$(DATADIR)/tc; \
 	done
 
 clean:
diff --git a/tc/q_netem.c b/tc/q_netem.c
index 90b2613..00df8a9 100644
--- a/tc/q_netem.c
+++ b/tc/q_netem.c
@@ -131,7 +131,7 @@ static int get_distribution(const char *type, __s16 *data, int maxdata)
 	char *line = NULL;
 	char name[128];
 
-	snprintf(name, sizeof(name), "%s/%s.dist", get_tc_lib(), type);
+	snprintf(name, sizeof(name), "%s/%s.dist", get_tc_datadir(), type);
 	f = fopen(name, "r");
 	if (f == NULL) {
 		fprintf(stderr, "No distribution data for %s (%s: %s)\n",
diff --git a/tc/tc_util.c b/tc/tc_util.c
index cf89fb7..052d44d 100644
--- a/tc/tc_util.c
+++ b/tc/tc_util.c
@@ -26,6 +26,10 @@
 #define LIBDIR "/usr/lib"
 #endif
 
+#ifndef DATADIR
+#define DATADIR "/usr/share"
+#endif
+
 static struct db_names *cls_names;
 
 #define NAMES_DB_USR CONF_USR_DIR "/tc_cls"
@@ -75,6 +79,17 @@ const char *get_tc_lib(void)
 	return lib_dir;
 }
 
+const char *get_tc_datadir(void)
+{
+	const char *data_dir;
+
+	data_dir = getenv("TC_DATA_DIR");
+	if (!data_dir)
+		data_dir = DATADIR "/tc/";
+
+	return data_dir;
+}
+
 int get_qdisc_handle(__u32 *h, const char *str)
 {
 	unsigned long maj;
diff --git a/tc/tc_util.h b/tc/tc_util.h
index ec20637..9474c53 100644
--- a/tc/tc_util.h
+++ b/tc/tc_util.h
@@ -72,6 +72,7 @@ struct exec_util {
 };
 
 const char *get_tc_lib(void);
+const char *get_tc_datadir(void);
 
 const struct qdisc_util *get_qdisc_kind(const char *str);
 const struct filter_util *get_filter_kind(const char *str);
